posthog/hogql_queries/insights/trends/aggregation_operations.py

```diff
@@ -281,6 +282,17 @@ def _actors_inner_select_query(
 
             return query
 
+        lookahead = ast.Call(name="toIntervalDay", args=[ast.Constant(value=1)])
+        if self.series.math in (
+            BaseMathType.WEEKLY_ACTIVE,
+            BaseMathType.MONTHLY_ACTIVE,
+        ) and self.query_date_range.interval_type in (IntervalType.WEEK, IntervalType.MONTH):
+            lookahead = (
+                ast.Call(name="toIntervalDay", args=[ast.Constant(value=7)])
+                if self.query_date_range.interval_type == IntervalType.WEEK
+                else ast.Call(name="toIntervalDay", args=[ast.Constant(value=30)])
```

Comment:
for month is it not possible to add a month interval directly instead of adding 30 days?
Not sure if it's still accurate when add 30 instead of 31?

Reply:
Trying to wrap my head around this again 😅  The WAU/MAU are defined as rolling 7 and 30 day window, so that's why we need these values here. You can also see this in the actors modal queries.

However there is still a bug here - the ternary condition should be based on the series math, not the interval type i.e. `self.series.math == BaseMathType.WEEKLY_ACTIVE` instead of `self.query_date_range.interval_type == IntervalType.WEEK`.

Thanks for catching!

Reply:
You were right all along, adding a month interval directly is the correct thing here. This is because we want the union of WAU from start of the month to end of the month. I've adapted the code accordingly. Also I added a change that swaps `<=` and `<` for start/end dates to be in line with what the actors query does.